src/mgr/DaemonHealthMetricCollector.cc

```diff
@@ -11,6 +11,20 @@ ostream& operator<<(ostream& os,
   return os << daemon.first << "." << daemon.second;
 }
 
+ostream& operator<<(
+  ostream& os,
+  const vector<DaemonHealthMetricCollector::DaemonKey>& daemons) {
```

Comment:
1. instead of repeating the implementation of `std::ostream& operator<<(std::ostream& out, const std::vector<A,Alloc>& v)`, i think we could take the opportunity to refactor it a little bit.
2. also, i think this commit and change itself deserve some comment
3. please put `{` in a new line.

like:
```c++
// define operator<<(ostream&, const vector<DaemonKey>&) after
// ostream& operator<<(ostream&, const DaemonKey&), so that C++'s
// ADL can use the former instead of using the generic one:
// operator<<(ostream&, const std::pair<A,B>&)
ostream& operator<<(
   ostream& os,
   const vector<DaemonHealthMetricCollector::DaemonKey>& daemons)
{
  os << "[";
  copy(std::begin(daemons), std::end(daemons),
       std::experimental::make_ostream_joiner(os, ","));
  os << "]";
  return os;
}
```

Reply:
I like it